control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go

```diff
@@ -431,3 +434,29 @@ func isOVNSBDBExposedThroughHCPRouter(hcp *hyperv1.HostedControlPlane) bool {
 
 	return util.IsPublicKASWithDNS(hcp) && publishingStrategy.Route.Hostname != ""
 }
+
+func SetRestartAnnotationAndPatch(ctx context.Context, crclient client.Client, dep *appsv1.Deployment, c config.DeploymentConfig) error {
+	if c.AdditionalAnnotations[hyperv1.RestartDateAnnotation] == "" {
+		return nil
+	}
+
+	if err := crclient.Get(ctx, client.ObjectKeyFromObject(dep), dep); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed retrieve deployment: %w", err)
+	}
+
+	patch := dep.DeepCopy()
+	podMeta := patch.Spec.Template.ObjectMeta
+	if podMeta.Annotations == nil {
+		podMeta.Annotations = map[string]string{}
+	}
+	podMeta.Annotations[hyperv1.RestartDateAnnotation] = c.AdditionalAnnotations[hyperv1.RestartDateAnnotation]
```

Comment:
Will this get rereconciled and squashed by the cluster-network-operator?

Reply:
Tested this change--can confirm the `multus-admission-controller` pods restart successfully now when triggered with the HC `restart-date` annotation. Also cycled the `cluster-network-operator` pod to verify it won't squash the restart annotation on the `multus-admission-controller` deployment, and that does seem to be the case.

Reply:
Awesome!!!! Approved